src/main/java/com/codeborne/selenide/Condition.java

```diff
@@ -771,4 +779,166 @@ public String getName() {
   public boolean missingElementSatisfiesCondition() {
     return missingElementSatisfiesCondition;
   }
+
+  /**
+   * Assert that element contains "value" attribute with date value that is satisfied to the pattern and is equal to expected date
+   *
+   * <p>Sample: {@code $("input").shouldHave(date(expectedLocalDate, "dd/MM/yyyy"));}</p>
+   *
+   * @param expectedDate expected value of "value" attribute
+   * @param pattern date format of value of "value" attribute
+   */
+  @CheckReturnValue
+  @Nonnull
+  public static Condition date(LocalDate expectedDate, String pattern) {
```

Comment:
I am thinking about adding so many overloaded methods to `Condition`. 
Probably we could add lesser number of those overloaded method by using parameter object like `ClickOptions`?

Reply:
I also thought about it but didn't find any good-looking solution.

The `Condition` class is like an entry point, where you can find all `pre-build` conditions. You just type `Condition.` and IDE shows all the possible variants. I think, that it's a good idea to have all condition in one place.


---

The best idea, that I have was intoducing some complex Condition, like

```Java
 class DateCondition {
   ...
   DateCondition withFormat(...)
   DateCondition withDate(...)
 }
 ```
 and then `public static DateCondition date( ?what is here, only date? )`.
 
 But, in my opinion, it's a little bit contra-intuitive, that there is a condition, that needs to be set up before use.
 And still there must be methods for `date`, `datetime`, `dateBetween`, `datetimeBetween`, `dateFormat` and `datetimeFormat`. It's just 4 methods lesser then with overloads...

Reply:
@asolntsev 
Oops.

Did you mean something like `public static Condition date(DateConditionOptions)`, where 
```Java
class DateConditionOptions {
   public static DateConditionOptions of(...) // common method, maybe with overloads 
   
   public static DateConditionOptions format(...) // for format check only
   
   public static DateConditionOptions beteween(DateConditionOptions, DateConditionOptions)
   
   ....
   
   public DateConditionOptions withFormat(...) //to be able to change format
}
```